src/TabBar.js

```diff
@@ -27,68 +27,61 @@ const TabButton = styled.TouchableOpacity`
     align-items: center;
     border-radius: 100;
     padding-vertical: 10;
-    flex-grow: ${p => p.isRouteActive ? (p.labelLength / 10 + 1) : 1};
-`
+    flex-grow: ${(p) => (p.isRouteActive ? (p.labelLength / 10 + 1) : 1)};
+`;
 
 const Label = styled(Animated.Text)`
-    color: ${p => p.activeColor};
+    color: ${(p) => p.activeColor};
     font-weight: bold;
-    margin-left: ${p => p.icon ? 8 : 0};
-`
+    margin-left: ${(p) => (p.icon ? 8 : 0)};
+`;
 
 const Dot = styled(Animated.View)`
     position: absolute;
-    top: ${p => p.topPadding};
-    width: ${p => p.width};
-    height: ${p => p.height};
+    top: ${(p) => p.topPadding};
+    width: ${(p) => p.width};
+    height: ${(p) => p.height};
     border-radius: 100;
-    background-color: ${p => p.activeTabBackground};
+    background-color: ${(p) => p.activeTabBackground};
     z-index: -1;
-`
+`;
 
 export default class TabBar extends React.Component {
-
     constructor(props) {
         super(props);
         this.state = {
-            prevPos: this.props.verticalPadding,
+            prevPos: props.verticalPadding,
             pos: 0,
             width: 0,
             height: 0,
             animatedPos: new Animated.Value(1),
+        };
+    }
+
+    componentDidUpdate(prevProps) {
+        const { navigation } = this.props;
+        const { animatedPos, pos } = this.state;
+        const { index: activeRouteIndex } = navigation.state;
+        const { index: prevActiveRouteIndex } = prevProps.navigation.state;
+
+        if (activeRouteIndex !== prevActiveRouteIndex) {
+            animatedPos.setValue(0);
+            this.setState({ prevPos: pos });
+            this.animation(animatedPos).start();
         }
     }
 
     animation = (value) => Animated.spring(value, {
         toValue: 1,
     })
 
-  componentDidMount() {
-    this.animation(this.state.animatedPos).start(() => {
-      this.setState({
-        prevPos: this.props.verticalPadding,
-      });
-      this.state.animatedPos.setValue(0);
-    });
-    if(Platform.OS === 'android'){
-        BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
+    onLayoutTabButton = (event, isRouteActive) => {
+        isRouteActive && this.setState({
+            pos: event.nativeEvent.layout.x,
+            width: event.nativeEvent.layout.width,
+            height: event.nativeEvent.layout.height
+        });
     }
-  }
-
-  componentWillUnmount() {
-    if(Platform.OS === 'android'){
-        BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
-    }
-  }
-
-  handleBackPress = () => {
-    this.animation(this.state.animatedPos).start(() => {
-      this.setState({
-        prevPos: this.state.pos,
-      });
-      this.state.animatedPos.setValue(0);
```

Comment:
Why did you remove this section? There was a bug in android with the physical back button.